### ISSUE TITLE
Error TS4020 or TS4023, working with mixins, base class has static symbol property, only when multiple files.

### ISSUE THREAD

**Issue by miguel-leon** [microsoft/TypeScript#52644]
# Bug Report

In the following code (file: `index.ts`)

```ts
export class TestComponent extends MixinComponent(Component<HTMLElement>)<number> {
}
```
I get error `TS4020: 'extends' clause of exported class 'TestComponent' has or is using private name 'Test'.`

I've gotten that error before and managed to fix it by exporting the *"private name"* the compiler is complaining about.

But in this case, said name **IS** being exported (file: `component.decl.ts`)
```ts
export const Test = Symbol("test");  // <--- HERE

export declare abstract class Component<T extends HTMLElement = HTMLElement> {
  readonly element: T;

  public static [Test]: string;
}
```

What other reason is there to cause this error? I don't see no *"private name"*.

It's also a different error `Exported variable 'MixinParent' has or is using name 'Test' from external module "/sandbox/component.decl" but cannot be named.` if I put the inheritance in two separate lines (file: `index.ts`)

```ts
export const MixinParent = MixinComponent(Component<HTMLElement>);

export class AnotherTestComponent extends MixinParent<string> {
}
```

### 🔎 Search Terms

all in title.

### 🕗 Version & Regression Information

latest.
(`declaration: true`)

### ⏯ Playground Link

[codesandbox.io link](https://codesandbox.io/s/typescript-playground-export-forked-r6enb3?file=/index.ts)

in playground is not reproducible, I suppose because everything is in a single file, but here's a link anyway.

[Playground link](https://www.typescriptlang.org/play?ts=4.9.5#code/KYDwDg9gTgLgBDAnmYcCyBLEGB2AeAKAEgBhAGwEMBnKuUGYHAE1ooCMqYoKBjeHYAHc4ACgB0EilADmVAFxwKORAG0AugEo4AXgB8i5ToOIANMQCCM2vUYtj6o0tVqC+7XACSAM3McuvGEIiUkoaM2D2Tm4+OAFBcUkrBUtZLT1PH2U8cmoqE2N8jxxOJR5gABVkYGzQql1dcKI4hLEpWWSrNP1vcyycsILPYphSiqqa3PriXWIAMgzexAmBgFccAGscCEEcfIAFDB515by4deBECC84frrdAG4CJ6QUBazy-PKoFeB8gDEKGQqMA3HAVE46CAGMxaOU4AB+WLAABuwCgcAUTjUkOhdhUAlRUGxiK+PwxcABQOAj2eVQWfmigVuONsrAZASRwhabXk9k0On0Tk+31+FMBwLcxGZNhhnNEElaST5XWMCLFVPJpOpTwIoEgsDgPAgwzg5WAnCMAGVEABbNgQMgiABEDE4To0NL10HgTGAPEoUFQkX8MX9uTgAH0SBAbZABDhAnCZXYABLlNAAGQAomRgDbGPB3GnMzm8wX9ABvAhwOCBihMY1kRB0XP5hMKco0mtgFZsMiHOAlGADlRmzhqBRRXDSR4AXx1XoNL1Q0djxoLeCTUNZcGL2dbBaMe9LbZgoKjMbjG-KDyei-gRpNq6v7YQVSukef64TRi8az4GDGqI1YIAAFhgvJOAQWhVrOiisMonrgN6b6vF+8YwCeG4kKCJAsrKwaMnK3JKk46gquhG64F4aKmvoJLkgSaI0neyEGn+OAAUBmDYDglEJngABC1CoMmtA8bgeDLh+-FnroIhsCJCjCcCGgKBJfGXt+MDRsM3x8NAQkiZWxCBjAKxQDghq1OgWC4LJ+F2ApInwcYMHEHB1DGHOrH6j6foBkG7KhjZEYabJhA1iQZg1luuK0MeB4Jq4jm0BFN5wFWNY8AoJA+b5KHLrZvGybpUQrAZUBGcCqXFZJ0nXLJ9Q6MQIiERycRwNkny1YlZY-u4slYQJKkgvJCo8rlxrlZVexSBQ+YMFAVDVSCKphXZmlrhh3V0XA8xFCUnFjCgq26FoszEAcRxSe+1wbSVWkYfk5yXNcDWRuFT3lvtV2HMco0vRcH6jbeOq6mxD42WOOnfT+Yl1VtL4wCIEV9ae514DgKx2miJnzgVBqPhaGlzYGA2I7JqNw4E6Plh6hNQ+GvQQDAoFojDDkI6TUgblOODSPjBBAA)

### 💻 Code

above.

### 🙁 Actual behavior

errors.

### 🙂 Expected behavior

no errors.

**Comment by patroza** [comment]
@RyanCavanaugh I have this issue here https://github.com/patroza/ts-repro-symbol-class/blob/main/file2.ts#L9

**Comment by evelant** [comment]
Another minimal reproduction: https://stackblitz.com/edit/github-a4rrlb?file=README.md

**Comment by justinfagnani** [comment]
Does anyone have a workaround for this? I've tried re-exporting the offending name from the consuming module, but no luck.
